server/lib/report_server/post_processing/job.ex

```diff
@@ -24,9 +24,15 @@ defmodule ReportServer.PostProcessing.Job do
     end
   end
 
+  # NOTE: if you add another preprocessing action here, this should be refactored to
+  # put the preprocessing in each step instead of it all being here - the learner
+  # preprocessing should be moved to a helper module that could be imported into the step
+  # so that it could be used for multiple steps
   defp preprocess_rows(job, query_result, overrides) do
     preprocessed = %{
-      learners: %{}
+      learners: %{},
+      merged_user_ids: %{},
+      user_resources: %{},
```

Comment:
Could you document the structures that are expected to end up in each of these maps? The code is difficult to follow to understand what is happening in these preprocessing steps, and what the expected output might be.